x/market/client/cli/flags.go

```diff
@@ -0,0 +1,45 @@
+package cli
+
+import (
+	"bytes"
+	"encoding/json"
+	flag "github.com/spf13/pflag"
+)
+
+const (
+	FlagDealFile = "deal-json-file"
+)
+
+type createDealInputs struct {
+	DataSchema            []string `json:"data_schema"`
+	Budget                string   `json:"budget"`
+	WantDataCount         uint64   `json:"want_data_count"`
+	TrustedDataValidators []string `json:"trusted_data_validators"`
+}
+
+type XCreateDealInputs createDealInputs
+
+type XCreateDealInputsExceptions struct {
+	XCreateDealInputs
+	Other *string
+}
```

Comment:
I wonder why X is attached to prefix.
Is this auto-generated?

Reply:
Oh, The prefix X is not auto-generated. I refer to [Osmosis's parse.go](https://github.com/osmosis-labs/osmosis/blob/main/x/gamm/client/cli/parse.go) and I just use X prefix.

Reply:
IMHO, we don't need `XCreateDealInputsExceptions`. I just looked at Osmosis source codes, and it seems that the `Other` field is for copying properties of an other existing pool in order to create a new pool. In our case, we aren't gonna need this `Other` option.